### PR TITLE
#3 added filter to elastic search appender

### PR DESCRIPTION
**Pull request by thaarbach**
Don't know if it works

### CODE REVIEW

**Comment by rfoltyns** [comment]
Looks good. I'll have another look later to see how we can test it.

File: log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/ElasticsearchAppender.java [hunk 1]
```diff
@@ -73,7 +75,7 @@ public static Builder newBuilder() {
         return new Builder();
     }
 
-    public static class Builder implements org.apache.logging.log4j.core.util.Builder<ElasticsearchAppender>  {
+    public static class Builder extends AbstractFilterable.Builder implements org.apache.logging.log4j.core.util.Builder<ElasticsearchAppender>  {
```
Comment (rfoltyns) [inline log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/ElasticsearchAppender.java]: I'm not a fan of extending `AbstractFiterable.Builder` here. `AbstractAppender.Builder` looks more appropriate, but let's defer this decision now.

File: log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/ElasticsearchAppender.java [hunk 2]
```diff
@@ -105,7 +107,7 @@ public ElasticsearchAppender build() {
                 layout = JsonLayout.newBuilder().setCompact(true).build();
             }
 
-            return new ElasticsearchAppender(name, layout, ignoreExceptions, batchDelivery, messageOnly);
+            return new ElasticsearchAppender(name, layout, getFilter(), ignoreExceptions, batchDelivery, messageOnly);
```
Comment (rfoltyns) [inline log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/ElasticsearchAppender.java]: I've added `filter` to this builder due to reasons stated above.

**Comment by rfoltyns** [comment]
Although it's possible to write a test for this, it would be like reinventing a wheel as filtering logic is in Log4j's scope. I'll apply some minor changes and push your commit today.

File: log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/ElasticsearchAppender.java (hunk 1 above)
Comment (thaarbach) [reply log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/ElasticsearchAppender.java]: Yes, looks better.

**Comment by thaarbach** [comment]
Looks good to me.

**Pull request closed without merging**